docs/spec/concepts.rst

```diff
@@ -3,97 +3,216 @@
-          return 1
-      else:
-          return x * 2
-
-To allow precise typing in such situations, the user should use
-a union type in conjunction with the ``enum.Enum`` class provided
-by the standard library, so that type errors can be caught statically::
-
-  from enum import Enum
-
-  class Empty(Enum):
-      token = 0
-  _empty = Empty.token
-
-  def func(x: int | None | Empty = _empty) -> int:
-
-      boom = x * 42  # This fails type check
-
-      if x is _empty:
-          return 0
-      elif x is None:
-          return 1
-      else:  # At this point typechecker knows that x can only have type int
-          return x * 2
-
-Since the subclasses of ``Enum`` cannot be further subclassed,
-the type of variable ``x`` can be statically inferred in all branches
-of the above example. The same approach is applicable if more than one
-singleton object is needed: one can use an enumeration that has more than
-one value::
-
-  class Reason(Enum):
-      timeout = 1
-      error = 2
-
-  def process(response: str | Reason = '') -> str:
-      if response is Reason.timeout:
-          return 'TIMEOUT'
-      elif response is Reason.error:
-          return 'ERROR'
-      else:
-          # response can be only str, all other possible values exhausted
-          return 'PROCESSED: ' + response
+Static, dynamic, and gradual typing
+-----------------------------------
+
+A **statically-typed** programming language runs a type-checker before running
+a program. The program is required to be well-typed according to the language's
+type system. The type system assigns a type to all expressions in the language
+and verifies that their uses obey the typing rules. Normally, a program that is
+not well-typed (i.e., one that contains a type error) will not run. Java and
+C++ are examples of statically-typed object-oriented languages.
+
+A **dynamically-typed** programming language does not run a type-checker before
+running a program. Instead, it will check the types of values before performing
+operations on them at runtime. This is not to say that the language is
+"untyped". Values at runtime have a type and their uses obey typing rules. Not
+every operation will be checked, but certain primitive operations in the
+language such as attribute access or arithmetic will be. Python was
+historically a dynamically-typed language.
+
+**Gradual typing** is the name for a specific way to combine static and dynamic
+typing. Type-annotated Python allows mixing static and dynamic checking at a
+fine level of granularity. Individual variables, parameters, and returns can
+optionally be given a static type. Data structures such as objects can have a
+mix of static and dynamic-only checking. For example, a dictionary could be
+annotated to have static checking of the key type but only dynamic checking of
+the value type.
+
+In gradual typing, a dynamically typed term is indicated by a special "unknown"
```

Comment:
The word "term" is often used in programming languages literature, more or less synonomously with "expression." See e.g. https://cs.stackexchange.com/a/64702

If the word doesn't communicate well, I'll find alternatives.

From the perspective of a static type checker, I think it makes sense to refer to things typed as Any as "dynamically checked," because the static type checker generally isn't able to check them, and thus the program is relying solely on dynamic checking of those values. (The fact that all values are dynamically checked isn't really relevant from the perspective of the static checker.) Perhaps the clarification that this is "from the perspective of the static checker" is missing. I'll see how I can rephrase.

Reply:
I've avoided the word "term", and adjusted quite a bit of wording. Curious if you like the new version better.